tests/functional/feature_flags/test_time_based_actions.py

```diff
@@ -0,0 +1,476 @@
+        environment="test_env",
+        application="test_app",
+        name="test_conf_name",
+        max_age=600,
+        sdk_config=Config(region_name="us-east-1"),
+    )
+    feature_flags: FeatureFlags = FeatureFlags(store=app_conf_fetcher)
+
+    # Evaluate our feature flag
+    context = {} if context is None else context
+    return feature_flags.evaluate(
+        name="my_feature",
+        context=context,
+        default=False,
+    )
+
+
+def test_time_based_utc_in_between_time_range_rule_match(mocker):
+    assert evaluate_mocked_schema(
+        mocker=mocker,
+        rules={
+            "lambda time is between UTC 11:11-23:59": {
+                RULE_MATCH_VALUE: True,
+                CONDITIONS_KEY: [
+                    {
+                        CONDITION_ACTION: RuleAction.SCHEDULE_BETWEEN_TIME_RANGE.value,  # this condition matches
+                        CONDITION_KEY: TimeKeys.CURRENT_TIME.value,
+                        CONDITION_VALUE: {TimeValues.START.value: "11:11", TimeValues.END.value: "23:59"},
+                    },
+                ],
+            }
+        },
+        mocked_time=(2022, 2, 15, 11, 12, 0, datetime.timezone.utc),
+    )
+
+
+def test_time_based_utc_in_between_time_range_no_rule_match(mocker):
+    assert not evaluate_mocked_schema(
+        mocker=mocker,
+        rules={
+            "lambda time is between UTC 11:11-23:59": {
+                RULE_MATCH_VALUE: True,
+                CONDITIONS_KEY: [
+                    {
+                        CONDITION_ACTION: RuleAction.SCHEDULE_BETWEEN_TIME_RANGE.value,  # this condition matches
+                        CONDITION_KEY: TimeKeys.CURRENT_TIME.value,
+                        CONDITION_VALUE: {TimeValues.START.value: "11:11", TimeValues.END.value: "23:59"},
+                    },
+                ],
+            }
+        },
+        mocked_time=(2022, 2, 15, 7, 12, 0, datetime.timezone.utc),  # no rule match 7:12 am
+    )
+
+
+def test_time_based_between_time_range_rule_timezone_match(mocker):
+    timezone_name = "Europe/Copenhagen"
+
+    assert evaluate_mocked_schema(
+        mocker=mocker,
+        rules={
+            "lambda time is between UTC 11:11-23:59, Copenhagen Time": {
+                RULE_MATCH_VALUE: True,
+                CONDITIONS_KEY: [
+                    {
+                        CONDITION_ACTION: RuleAction.SCHEDULE_BETWEEN_TIME_RANGE.value,  # this condition matches
+                        CONDITION_KEY: TimeKeys.CURRENT_TIME.value,
+                        CONDITION_VALUE: {
+                            TimeValues.START.value: "11:11",
+                            TimeValues.END.value: "23:59",
+                            TimeValues.TIMEZONE.value: timezone_name,
+                        },
+                    },
+                ],
+            }
+        },
+        mocked_time=(2022, 2, 15, 11, 11, 0, gettz(timezone_name)),  # rule match 11:11 am, Europe/Copenhagen
+    )
```

Comment:
We should add a new test contemplating intervals between 11:00 and 13:00, for example.